array-reduce/main.js

```diff
@@ -0,0 +1,47 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+// Sum all numbers
+const sumAllNumbers = (total, num) => {
+  return total + num;
+};
+console.log('Sum all numbers: ', numbers.reduce(sumAllNumbers, 0));
```

Comment:
It doesn't hurt anything but it isn't necessary to pass `0` as the second argument here